core/src/main/java/io/undertow/server/DefaultByteBufferPool.java

```diff
@@ -139,6 +139,7 @@ public PooledByteBuffer allocate() {
             buffer = queue.poll();
             if (buffer != null) {
                 currentQueueLengthUpdater.decrementAndGet(this);
+                buffer.clear();
```

Comment:
This doesn't clear all buffers created / reused in this method, see below comment.